tests/0081-admin.c

```diff
@@ -3308,6 +3308,207 @@ static void do_test_DescribeConsumerGroups(const char *what,
+
+        /* Create the topics first. */
+        test_CreateTopics_simple(rk, NULL, &topic, 1, partitions_cnt, NULL);
+
+        /* Verify that topics are reported by metadata */
+        test_wait_metadata_update(rk, &exp_mdtopic, 1, NULL, 0, 15 * 1000);
+
+        /* Produce 100 msgs */
+        test_produce_msgs_easy(topic, testid, 0, msgs_cnt);
+
+        for (i = 0; i < TEST_DESCRIBE_CONSUMER_GROUPS_CNT; i++) {
+                rd_kafka_conf_t *conf;
+                char *group_id = rd_strdup(test_mk_topic_name(__FUNCTION__, 1));
+                if (i < 2) {
+                        /* Classic Protocol */
+                        test_conf_init(&conf, NULL, 0);
+                        test_conf_set(conf, "group.protocol", "classic");
+                } else {
+                        /* Consumer Protocol */
+                        test_conf_init(&conf, NULL, 0);
+                        test_conf_set(conf, "group.protocol", "consumer");
+                }
+
+                snprintf(client_ids[i], sizeof(client_ids[i]),
+                         "client_id_%" PRId32, i);
+
+                test_conf_set(conf, "client.id", client_ids[i]);
+                test_conf_set(conf, "session.timeout.ms", "5000");
+                test_conf_set(conf, "auto.offset.reset", "earliest");
+                rks[i] = test_create_consumer(group_id, NULL, conf, NULL);
+                test_consumer_subscribe(rks[i], topic);
+                /* Consume messages */
+                test_consumer_poll("consumer", rks[i], testid, -1, -1, msgs_cnt,
+                                   NULL);
+
+                expected[i].group_id = group_id;
+                expected[i].err      = RD_KAFKA_RESP_ERR_NO_ERROR;
+                describe_groups[i]   = group_id;
+        }
+
+        TIMING_START(&timing, "DescribeConsumerGroups");
+        TEST_SAY("Call DescribeConsumerGroups\n");
+        rd_kafka_DescribeConsumerGroups(
+            rk, describe_groups, TEST_DESCRIBE_CONSUMER_GROUPS_CNT, options, q);
+        TIMING_ASSERT_LATER(&timing, 0, 50);
+
+        TIMING_START(&timing, "DescribeConsumerGroups.queue_poll");
+
+        /* Poll result queue for DescribeConsumerGroups result. */
+        while (1) {
+                rkev = rd_kafka_queue_poll(q, tmout_multip(20 * 1000));
+                TEST_SAY("DescribeConsumerGroups: got %s in %.3fms\n",
+                         rd_kafka_event_name(rkev),
+                         TIMING_DURATION(&timing) / 1000.0f);
+                if (rkev == NULL)
+                        continue;
+                if (rd_kafka_event_error(rkev))
+                        TEST_SAY("%s: %s\n", rd_kafka_event_name(rkev),
+                                 rd_kafka_event_error_string(rkev));
+
+                if (rd_kafka_event_type(rkev) ==
+                    RD_KAFKA_EVENT_DESCRIBECONSUMERGROUPS_RESULT) {
+                        break;
+                }
+
+                rd_kafka_event_destroy(rkev);
+        }
```

Comment:
We can extract this functionality into a method in the same file or in `test.c` file which can be used by other functions in this test file.